circuit-codegen/src/main/kotlin/com/slack/circuit/codegen/CircuitSymbolProcessorProvider.kt

```diff
@@ -225,14 +236,27 @@ private class CircuitSymbolProcessor(
           fd.assistedParameters(symbols, logger, screenKSType, factoryType == FactoryType.PRESENTER)
         codeBlock =
           when (factoryType) {
-            FactoryType.PRESENTER ->
+            FactoryType.PRESENTER -> {
+              if (!name.endsWith("Presenter")) {
+                logger.error(
+                  "CircuitPresenter class names must end in 'Presenter'.",
+                  annotatedElement
+                )
+              }
               CodeBlock.of(
                 "%M·{·%M(%L)·}",
                 MemberName("com.slack.circuit", "presenterOf"),
                 MemberName(packageName, name),
                 assistedParams
               )
+            }
             FactoryType.UI -> {
+              if (!name.endsWith("Ui")) {
```

Comment:
Similar to my comment above, we could make this more precise (without imposing a naming convention) by verifying that the state consuming function:
1. accepts an instance of `CircuitUiState`
2. has no other arguments (that don't provide a default value)
3. returns Unit